compiler-rt/test/asan/TestCases/Windows/dll_operator_array_new_with_dtor_left_oob.cpp

```diff
@@ -2,12 +2,18 @@
 // RUN: %clang_cl_asan %LD %Od %s %Fe%t.dll
 // RUN: not %run %t %t.dll 2>&1 | FileCheck %s
 
+#include "../defines.h"
+
 struct C {
   int x;
   ~C() {}
 };
 
-int __attribute__((noinline, optnone)) hide(int x) { return x; }
+int ATTRIBUTE_NOINLINE
+#if __clang__ || !defined(_MSC_VER)
```

Comment:
I guess this should be `defined(__clang__)` instead of plain `__clang__`?

Reply:
I think this is somewhat stylistic. I can switch to defined instead of relying on the substitution of undefined identifiers for 0 though, it's a reasonable style.

Reply:
Yeah I think we in general would prefer to use `defined()` rather than relying on expansion to zero (although I'm not sure if we have a written down policy about it), so the code would be clean for `-Wundef` too.